### ISSUE TITLE
Extension does not open review/lesson page

### ISSUE THREAD

**Issue by simonyangme** [wiiaboo/Wanikani-Notifier#4]
When there are reviews available, clicking on the extension does not open the review page. Right clicking the extension and clicking the review/lesson options also do nothing.

Also, thanks for the very quick release to support the v2 API! Been slacking on WK until I updated the extension today 😅

**Comment by wiiaboo** [comment]
Can you try going to the Options page and clicking "Reset"? (copy your API Token before resetting or copy it again from Wanikani to make sure it's valid)
I didn't take too much care on backwards compatibility with older stored settings, so older installations might stay broken until everything is reset.

**Comment by simonyangme** [comment]
Just reset and still same behavior. API token works as it does show the review count. Disabling-reenabling extension and restarting the browser didn't fix it either. Let me know if there's any other info I can provide to help debug the issue. I haven't done any extension dev before but can poke around with some guidance.

**Comment by wiiaboo** [comment]
Go to the URL `about:devtools-toolbox?type=extension&id=%40wanikani-notifier` (or "Manage Extension" through the extension button, then "Debug Addons" in the wheel icon, scroll to this extension then click "Inspect")
and take a screenshot of the Extension Storage values in the "Storage" tab there (censor the API key and maybe the username too, if you want).

Should look like this:
![Screenshot 2020-09-18 at 21 20 47](https://user-images.githubusercontent.com/111605/93641883-fff7fc80-f9f4-11ea-846a-e5895118e290.png)

**Comment by simonyangme** [comment]
![image](https://user-images.githubusercontent.com/48696/93646782-bf2fc180-f9bb-11ea-8a62-92fc54b9596b.png)

It also doesn't seem to auto-update for some reason. Right clicking and choosing refresh does update the review and lesson count though.

**Comment by wiiaboo** [comment]
Pressing "Save" should force the "user" to be updated, but that is only good for confirming the api_key works for now, you shouldn't really need it to be updated for things to work.

The "summary" information is only refreshed close to when the next review is available, or if there are already reviews available, every X minutes, in your case, every 15.
I should change it to at most 4 hours later, since that's the minimum review time for something you just learned.


I also haven't had time to pick up reviews again, so I haven't been able to test it myself.

I'll try to test myself this weekend.

**Comment by simonyangme** [comment]
I poked at the code a little bit, and I have a clue for some of the above issues.

The code around alarm seems to be faulty. `setAlarm` gets called with no argument in some cases, and gets called with a date in other instances, which is not valid.

I also figured out the issue with the page not opening. `openPageOnce` is a noop if the page is already open elsewhere, which doesn't work for me as I had an instance of the review/lesson page lost in my windows of tabs. At the very least it should bring that tab into focus, or just create a new tab.

**Comment by wiiaboo** [comment]
> I also figured out the issue with the page not opening. `openPageOnce` is a noop if the page is already open elsewhere, which doesn't work for me as I had an instance of the review/lesson page lost in my windows of tabs. At the very least it should bring that tab into focus, or just create a new tab.

Yeah, I just implemented it as not opening a new tab if some other with the right title is already open. I didn't remember that it would be good to focus it. Maybe because I don't really use WaniKani anymore myself 😅 

> The code around alarm seems to be faulty. `setAlarm` gets called with no argument in some cases, and gets called with a date in other instances, which is not valid.

It's called with an argument when there's a specific timestamp. If no argument is used, it defaults to the user's `update_interval`. i.e. it alarms when the next review is available or it alarms every X minutes.

Or at least that's how I meant it to work and that's how it worked while I tested it.

With the next version I was thinking of adding a Debug textarea in options to send pseudo-logs to that should be more helpful for issue reporting.

I haven't had time or much patience after work to investigate this recently, sorry.
